### PR TITLE
Switch memo flags to StringSlice

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Switch `memo_*` flags to `StringSlice`.

**Why?**
Memo flags are broken. For some reason `fastjson` parser was used for memo which can be arbitrary data and doesn't have to be JSON. Space was used a separator between keys and values which is not standrat way. This PR fixes memo flags and uses multiply flags with the same name to pass several keys/values.
Example:
```
tctl workflow start --taskqueue test --workflow_type some_type --memo_key TestKey --memo "test value" --memo_key TestKey2 --memo value2

```

**How did you test it?**
Existing tests and local run.

**Potential risks**
Memo flags was broken so it is very unlikely that anyone used it.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: tools/cli/flags.go [hunk 1]
```diff
@@ -362,30 +366,28 @@ func getFlagsForStart() []cli.Flag {
...
+			Usage: fmt.Sprintf("Optional key of memo. If there are multiple keys, provide multiple %s flags", FlagMemoKey),
 		},
-		cli.StringFlag{
+		cli.StringSliceFlag{
 			Name: FlagMemo,
-			Usage: "Optional info that can be showed when list workflow, in JSON format. If there are multiple JSON, concatenate them and separate by space. " +
-				"The order must be same as memo_key",
+			Usage: fmt.Sprintf("Optional info that can be showed when list workflow. If there are multiple values, provide multiple %s flags. "+
+				"The order must be same as %s", FlagMemo, FlagMemoKey),
 		},
```
Comment (yiminc) [inline tools/cli/flags.go:377]: I think it is worth to show an example of how to use with 2 keys and 2 values.

File: tools/cli/workflowCommands.go [hunk 2]
```diff
@@ -298,74 +297,69 @@ func formatInputsForDisplay(inputs []interface{}) string {
...
-	rawSearchAttrVals := sanitize(c.String(FlagSearchAttributesVal))
-	if len(rawSearchAttrVals) == 0 {
+	saKeys := c.StringSlice(FlagSearchAttributesKey)
+
+	var saValues []string
+	if c.IsSet(FlagSearchAttributesValue) {
+		saValues = c.StringSlice(FlagSearchAttributesValue)
+	} else if c.IsSet(FlagSearchAttributesKey) {
+		ErrorAndExit(fmt.Sprintf("Search attribute values must be provided using %s.", FlagSearchAttributesValue), nil)
+	} else {
```
Comment (yiminc) [inline tools/cli/workflowCommands.go:311]: These 2 cases should be combined with the first if check of this func to make this more readable.

```
isKeySet := c.IsSet(FlagSearchAttributesKey)
isValueSet := c.IsSet(FlagSearchAttributesValue)
if !isKeySet && !isValueSet {
  return nil
} else if isKeySet && !isValueSet {
  ErrorAndExit(...)
} else if !isKeySet && isValueSet {
  ErrorAndExit(...)
}
saKeys := c.StringSlice(FlagSearchAttributesKey)
saValues = c.StringSlice(FlagSearchAttributesValue)
...
```

File: tools/cli/workflowCommands.go [hunk 3]
```diff
@@ -298,74 +297,69 @@ func formatInputsForDisplay(inputs []interface{}) string {
...
+		if err != nil {
+			ErrorAndExit(fmt.Sprintf("Error reading memo file %s.", inputFile), err)
+		}
+		memoValues = strings.Split(string(data), "\n")
+	} else if c.IsSet(FlagMemo) {
+		memoValues = c.StringSlice(FlagMemo)
+	} else if c.IsSet(FlagMemoKey) {
+		ErrorAndExit(fmt.Sprintf("Memo values must be provided using %s or %s.", FlagMemo, FlagMemoFile), nil)
+	} else {
+		return nil
```
Comment (yiminc) [inline tools/cli/workflowCommands.go:358]: Same comment apply here, should do all these check at the beginning of this func and make it more easy to read.
```
isKeySet := ...
isMemoSet := ...
isFileSet := ...
isValueSet := isMemoSet || isFileSet...
if !isKeySet && !isValueSet {
  return nil
} else if isKeySet && !isValueSet {
  ErrorAndExit(...)
} else if !isKeySet && isValueSet {
  ErrorAndExit(...)
} else if isMemoSet && isFileSet {
  ErrorAndExit(...)
} 
memoKeys := c.StringSlice(FlagMemoKey)
memoValues := c.StringSlice(FlagMemo)
if isFileSet {
   memoValues = ...
}
...
```

File: tools/cli/flags.go (hunk 1 above)
Comment (alexshtin) [reply tools/cli/flags.go:377]: Added example to PR description. I don't want to overload help screen as multiply flags with the same name will be a common pattern in new `tctl` and supposed to be well known.

**Review by yiminc — APPROVED** [review]

**Pull request merged**